### ISSUE TITLE
ClientID Logging

### ISSUE THREAD

**Issue by kiwilad-nz** [AdguardTeam/AdGuardHome#5462]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

FreeBSD

### CPU architecture

AMD64

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

Version:  v0.107.23

#### What did you do?
I've added a clientID called mobile, added mobile as an allowed client under access settings and added a public DNS entry of mobile under my domain for public DoT/DoH access.
#### Expected result
The logs for mobile.mydomain.tld (mobile) show only the hosts/IP that use that ClientID.
#### Actual result
In the logs, I have cross over from local hosts showing up in the ClientID logs, those of which I know are not using the domain (mobile) ClientID.
#### Screenshots (if applicable)
![Logs](https://user-images.githubusercontent.com/50952231/217130635-c9724a39-35a2-4ff9-8a05-bef461d54e3c.PNG)
![AccessSettings](https://user-images.githubusercontent.com/50952231/217130647-9f6add7e-b4b8-40b6-a1c7-4a8ef6fd250a.PNG)
![ClientID](https://user-images.githubusercontent.com/50952231/217130701-594f28e7-8b75-49b7-a24c-c239bcc177a9.PNG)
![ClientID2](https://user-images.githubusercontent.com/50952231/217130707-b6f62d32-1353-4621-b944-f1114a7b026c.PNG)
![image](https://user-images.githubusercontent.com/50952231/217131481-c8fe39b7-1f83-4ed7-94f5-46d09c78568f.png)

#### Additional information
Adguard Home running on pfSense

**Comment by kiwilad-nz** [comment]
I think I have resolved this, although I think a confusing issue and still don't understand why this caused it to be problematic.

Under my encryption settings, for the domain, I did not have a FQDN entered but instead just the domain name, the cert for the domain was wildcarded however. This worked for the most part but I thought a wildcarded domain would allow all hosts to be used from this method. I however found this not to be the case.

For example:
domain.tld (worked)
dns.domain.tld (worked)
mobile.dns.domain.tld (failed)

I then changed the cert to be a wildcarded FQDN instead and the logging stopped becoming problematic, with me also changing the public dns entry to match mobile.dns.domain.tld instead of dns.domain.tld.

**Comment by kiwilad-nz** [comment]
Oddly enough, I think I have resolved this for myself and identified two problems with two ways to avoid the issues. I'm still waiting to see if the issues reappear but so far it seems fine.

The 1st issue  (the logs issue). I can't explain but have resolved this by using ClientID's that incorporate a "-", so in my example bills-mobile instead of using singular words.

I then set up a public wildcard subdomain DNS A record, wildcard probably helps for security reasons as I imagine the DNS entry that is required to access AGH is now obfuscated.

The 2nd issue I came into was rogue ClientID showing up in my logs even though the ClientID and associated ClientID under Access Settings was removed. To get around this, removing all Access List settings and readding them seems to of cleared this bug for me and prevent the rogue ClientID reappearing in the logs.
